awss3/s3_bucket_test.go

```diff
@@ -59,11 +61,12 @@ var publicPolicy = `{
 
 func TestCreate(t *testing.T) {
 	cases := []struct {
-		Name          string
-		BucketName    string
-		BucketDetails awss3.BucketDetails
-		Location      string
-		Error         error
+		Name                                string
+		BucketName                          string
+		BucketDetails                       awss3.BucketDetails
+		Location                            string
+		Error                               error
+		expectDeletePublicAccessBlockCalled bool
```

Comment:
I'm not normally in favor of this approach to testing with mocks; I think it changes the test from a black box test to a white box test by checking if the code does certain things internally. In particular I normally try to veto it when mocking database calls, where checks in mocks can quickly converge to "did the function run this exact SQL statement". That said, in this case I think it helps to debug the function by making sure we're taking the correct path in the code when creating a public bucket, so I think it should stay. (No action required; just wanted to reason through it.)

Reply:
The acceptance tests were failing because the code to delete the public access block was never being called (due to a typo). So I wanted to create a unit test to verify that this logic is called correctly for public bucket policies, since it's much easier to debug a unit test than an acceptance test. And since it's important behavior, it should be unit tested